### PR TITLE
Update pointcloud_io interface

### PR DESCRIPTION
**Pull request by willdunklin**
Updated use of pointcloud_io interface to follow standard practices. Fixed bug with pointcloud_io failing to save filepath that does not exist yet.

### CODE REVIEW

File: CMake/kwiver_config.h.in [hunk 1]
```diff
@@ -8,6 +8,4 @@
 #ifndef KWIVER_CONFIG_H
 #define KWIVER_CONFIG_H
 
-#cmakedefine KWIVER_ENABLE_PDAL
```
Comment (mleotta) [inline CMake/kwiver_config.h.in:11]: We can probably just delete this whole file now that it is empty and not used.

File: arrows/pdal/tests/test_pointcloud_io.cxx [hunk 2]
```diff
@@ -39,78 +41,75 @@ class pointcloud_io : public ::testing::Test
 // ----------------------------------------------------------------------------
 TEST_F(pointcloud_io, create)
 {
-  using namespace kwiver::vital;
-  plugin_manager::instance().load_all_plugins();
```
Comment (mleotta) [inline arrows/pdal/tests/test_pointcloud_io.cxx:43]: There is a reason to load the plugins here an not for all tests.  Other than this special "create" test, it's probably better to directly instantiate the pdal object in the other tests.  This way if there is a failure in the plugin loading it only impacts one test that is directly testing plugin loading and the other tests will still run.  There is no problem having the PDAL tests directly link to the PDAL arrow library.

File: arrows/pdal/tests/test_pointcloud_io.cxx [hunk 3]
```diff
@@ -39,78 +41,75 @@ class pointcloud_io : public ::testing::Test
...
+  static auto geo_conv = kwiver::arrows::proj::geo_conversion{};
+  kv::set_geo_conv(&geo_conv);
 
-    kwiver::arrows::pdal::pointcloud_io pc_io;
-    pc_io.set_local_geo_cs(lgcs);
-    pc_io.save(tmp_path, points, colors);
+  auto lgcs = kv::local_geo_cs();
+  read_local_geo_cs_from_file(lgcs, geo_origin_path);
+
+  auto pc_io = kv::algo::pointcloud_io::create("pdal");
```
Comment (mleotta) [inline arrows/pdal/tests/test_pointcloud_io.cxx:82]: See comment above.  Probably better to use direct instantiation for most tests.

File: arrows/pdal/tests/test_pointcloud_io.cxx [hunk 4]
```diff
@@ -39,78 +41,75 @@ class pointcloud_io : public ::testing::Test
...
 TEST_F(pointcloud_io, create)
 {
-  using namespace kwiver::vital;
-  plugin_manager::instance().load_all_plugins();
-
-  EXPECT_NE(nullptr, algo::pointcloud_io::create("pdal"));
+  EXPECT_NE(nullptr, kv::algo::pointcloud_io::create("pdal"));
 }
 
 TEST_F(pointcloud_io, save) {
```
Comment (mleotta) [inline arrows/pdal/tests/test_pointcloud_io.cxx:47]: I missed this style issue before.  The KWIVER style is to have each `{` and `}` on a new line.

File: arrows/pdal/tests/test_pointcloud_io.cxx [hunk 5]
```diff
@@ -39,78 +41,75 @@ class pointcloud_io : public ::testing::Test
...
-      points.push_back(lm.second->loc());
-      colors.push_back(lm.second->color());
+  auto const geo_origin_path = data_dir + "/geo_origin.txt";
+  auto const landmarks_path = data_dir + "/landmarks.ply";
+  auto const tmp_path = data_dir + "/pointcloud.las";
+  std::ofstream ofs(tmp_path);
+  ofs.close();
+
+  // This will delete the temporary file even if an exception is thrown
+  struct _tmp_file_deleter {
```
Comment (mleotta) [inline arrows/pdal/tests/test_pointcloud_io.cxx:55]: Put `{` on a new line

File: arrows/pdal/tests/test_pointcloud_io.cxx [hunk 6]
```diff
@@ -39,78 +41,75 @@ class pointcloud_io : public ::testing::Test
...
+    ~_tmp_file_deleter() {
```
Comment (mleotta) [inline arrows/pdal/tests/test_pointcloud_io.cxx:56]: Put `{` on a new line

File: arrows/pdal/tests/test_pointcloud_io.cxx [hunk 7]
```diff
@@ -39,78 +41,75 @@ class pointcloud_io : public ::testing::Test
...
-    pc_io.save(tmp_path, points, colors);
+  auto lgcs = kv::local_geo_cs();
+  read_local_geo_cs_from_file(lgcs, geo_origin_path);
+
+  auto pc_io = kv::algo::pointcloud_io::create("pdal");
+  pc_io->set_local_geo_cs(lgcs);
+  pc_io->save(tmp_path, points, colors);
 }
 
 TEST_F(pointcloud_io, save_landmarks) {
```
Comment (mleotta) [inline arrows/pdal/tests/test_pointcloud_io.cxx:87]: Put `{` on a new line

File: arrows/pdal/tests/test_pointcloud_io.cxx [hunk 8]
```diff
@@ -39,78 +41,75 @@ class pointcloud_io : public ::testing::Test
...
-        std::remove( tmp_path.c_str() );
-      }
+  auto const geo_origin_path = data_dir + "/geo_origin.txt";
+  auto const landmarks_path = data_dir + "/landmarks.ply";
+  auto const tmp_path = data_dir + "/pointcloud.las";
+  std::ofstream ofs(tmp_path);
+  ofs.close();
+
+  // This will delete the temporary file even if an exception is thrown
+  struct _tmp_file_deleter {
```
Comment (mleotta) [inline arrows/pdal/tests/test_pointcloud_io.cxx:95]: Put `{` on a new line

File: arrows/pdal/tests/test_pointcloud_io.cxx [hunk 9]
```diff
@@ -39,78 +41,75 @@ class pointcloud_io : public ::testing::Test
...
+    ~_tmp_file_deleter() {
```
Comment (mleotta) [inline arrows/pdal/tests/test_pointcloud_io.cxx:96]: Put `{` on a new line

**Review by mleotta — CHANGES_REQUESTED** [review]

File: arrows/vtk/applets/color_mesh.cxx [hunk 10]
```diff
@@ -480,14 +474,9 @@ class color_mesh::priv : public kwiver::arrows::vtk::mesh_coloration
...
-    kwiver::arrows::pdal::pointcloud_io pc_io;
-    pc_io.set_local_geo_cs(lgcs);
-    pc_io.save(output_path, points, colors);
-#else
-    throw vital::file_write_exception(output_path,
-                                      "KWIVER was not compiled with PDAL, "
-                                      "cannot write LAS.");
-#endif
+    auto pc_io = vital::algo::pointcloud_io::create("pdal");
+    pc_io->set_local_geo_cs(lgcs);
```
Comment (mleotta) [inline arrows/vtk/applets/color_mesh.cxx:478]: We need a check here that `pc_io` was created successfully and is not null before we try to use it.  You can thrown an exception if `pc_io` is null.

File: arrows/vtk/applets/fuse_depth.cxx [hunk 11]
```diff
@@ -418,14 +412,9 @@ class fuse_depth::priv
...
-        kwiver::arrows::pdal::pointcloud_io pc_io;
-        pc_io.set_local_geo_cs(lgcs);
-        pc_io.save(output_mesh_file, points);
-#else
-        throw vital::file_write_exception(output_mesh_file,
-                                          "KWIVER was not compiled with PDAL, "
-                                          "cannot write LAS.");
-#endif
+        auto pc_io = vital::algo::pointcloud_io::create("pdal");
+        pc_io->set_local_geo_cs(lgcs);
```
Comment (mleotta) [inline arrows/vtk/applets/fuse_depth.cxx:416]: We need a check here that `pc_io` was created successfully and is not null before we try to use it.  You can thrown an exception if `pc_io` is null.

**Review by mleotta — CHANGES_REQUESTED** [review]
Looks pretty good, but needs checking in a couple places that the `pc_io` was successfully constructed.

**Review by mleotta — APPROVED** [review]
Need to resolve conflicts now that the branch to relocate test data has landed, but otherwise looks good.

**Comment by mleotta** [comment]
jenkins test this please

**Comment by willdunklin** [comment]
jenkins test this please

**Comment by willdunklin** [comment]
jenkins test this please

**Comment by willdunklin** [comment]
jenkins test this please

**Comment by mleotta** [comment]
jenkins test this please

**Review by mleotta — APPROVED** [review]

**Pull request merged**